File: binary-search/search_insert_position/search_insert_position.cpp
#include<vector.h>



class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        int left = 0;
        int right = nums.size() - 1;
        int mid = 0;
        if(target < nums[left]){
            return 0;
        }

        if(target > nums[right]){
            return right + 1;
        }

        while(left < right){
            mid = left + int((right-left) / 2 );
            if(nums[mid] > target){
                right = mid;

            }
            else if(nums[mid]  < target){
                left = mid + 1;
            }
            else if(nums[mid] == target){
                return mid;
            }

        }
        return left;

    }
};

File: binary-search/search_insert_position/search_insert_position.py
class Solution:
    def searchInsert(self, nums: [int], target: int) -> int:
        left = 0
        right = len(nums)-1
        
        if target < nums[left]:
            return 0
        
        if target > nums[right]:
            return right + 1
        
        while(left < right):
            mid = (left + right)//2
            if nums[mid] > target:
                right = mid
            elif nums[mid] < target:
                left = mid + 1
            else:
                return mid

        return left
            